consent/migrations/0002_migrate_to_new_data_sharing_consent.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+"""
+Custom migration to transfer consent data from the ``enterprise`` application to the ``consent`` application's
+``DataSharingConsent`` model.
+
+This migration only does anything significant when running forward. If you would like to do a backwards-migration,
+delete all new rows that were populated in the ``DataSharingConsent`` model's tables.
+(If you have new data in the tables that weren't migrated, use timestamps as you see fit).
+
+The reason there's no backwards-migration in code is because we can't reasonably determine which
+data was transferred and which data is new at the time of the migration. In order to avoid deleting
+new data, we simply leave it to someone with DB access to manage things based on the desired timestamp.
+"""
+
+from __future__ import unicode_literals
+
+from consent.models import DataSharingConsent
+
+from django.db import migrations
+
+from enterprise.models import EnterpriseCourseEnrollment, UserDataSharingConsentAudit
+
+
+def populate_data_sharing_consent(apps, schema_editor):
```

Comment:
So I realized the reason for this duplication issue was my fault for not using the username... duh. But accessing the username when dynamically getting models, as well as checking UDSCA for backup requires a bit more code, so it's there now -- let me know if that looks reasonable.